sumcheck/src/prover_v2.rs

```diff
@@ -131,9 +133,8 @@ impl<'a, E: ExtensionField> IOPProverStateV2<'a, E> {
             }
 
             let mut prover_msgs = Vec::with_capacity(num_variables);
-            let thread_id = max_thread_id - 1;
             let mut prover_state = Self::prover_init_with_extrapolation_aux(
-                mem::take(&mut polys[thread_id]),
```

Comment:
Treating the last polynomial different seems a bit weird to me.  I wonder if we can find a better way to organise the threads.  (But this PR isn't making things worse, so it's fine.)